url054c08669f_jules2689_CSI4124_tgz-pJ8-SMThemeParkExperi3J8/src/simModel/DVPs.java

```diff
@@ -0,0 +1,111 @@
+package simModel;
+
+class DVPs {
+	SMThemePark model; // for accessing the clock
+
+	public DVPs(SMThemePark model) {
+		this.model = model;
+	}
+
+	protected double uLoadUnloadTime() {
+		if (model.boardingOption == Constants.DOUBLE_SIDED) {
+			return 1.25; // double sided
+		} else {
+			return 2.0; // single sided
+		}
+	}
+
+	protected double travelTime(int nextId) {
+		switch (nextId) {
+		case Constants.SH:
+			return 5.0;
+		case Constants.GI:
+			return 8.0;
+		case Constants.RC:
+			return 7.0;
+		default:
+			return 6.0;
+		}
+	}
+
+	protected double getPercentageOfCustomersLeaving(int fromStation, int toStation) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 74 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.